### PR TITLE
[QC-752] Facilitate switching between provenances in post-processing

### PR DESCRIPTION
**Pull request by knopers8**
This makes the `retrieveMO` and `retrieveQO` methods expect a path without provenance for the prefix. Instead, it adds it on its own according to what is indicated in the Activity.

The PR also adapts object paths in the user code. I did it in the same PR to avoid breaking the functionality for even a short time. I will merge once all the concerned parties agree.

### CODE REVIEW

**Comment by iravasen** [comment]
@knopers8 thanks for implementing this. If I understand correctly the thing that changes for us is the removal of "qc" in front of each path. And if one wants to change from qc to (for example) qc_async, you add a parameter to the json that specifies this? Thanks.

**Comment by knopers8** [comment]
@iravasen Yes, you change "provenance" in "Activity" from "qc" to "qc_async" instead of changing it everywhere in your json

**Comment by knopers8** [comment]
@wiechula , @njacazio Are you OK with this?

If we merge, we would have coordinate the corresponding changes in all config files at P2 as well.

**Comment by njacazio** [comment]
@knopers8 looks good to me

**Comment by wiechula** [comment]
@knopers8 also fine for me.

@stheckel and @tklemenz please also have a look.

**Comment by tklemenz** [comment]
@knopers8 Would it also affect things like the update trigger newObject path 
e.g. `"updateTrigger": [ "newobject:ccdb:qc/TPC/MO/Clusters/ClusterData" ]` 
where it then would change the path? This would be inconvenient imho.

**Comment by knopers8** [comment]
@tklemenz 
if you choose `ccdb` in `"newobject:ccdb:qc/TPC/MO/Clusters/ClusterData"`, then you shall indicate the full path and the framework won't add any prefix. This is because you mean to refer to CCDB, not QCDB, so not QC objects. In QC config files, CCDB is indicated in `conditionDB.url` parameter.
If you put `qcdb`, the provenance will be added according to what is in `Activity.provenance`, because we operate in the QCDB domain.
I am not sure I understood you there, is it actually convenient or not?

**Comment by tklemenz** [comment]
Thanks for the clarification! When there is a distinction between `ccdb` and `qcdb` in adding the prefix or not, as you explained, then it is absolutely fine. 

So to me, this also looks good.

**Comment by stheckel** [comment]
Dear Piotr, dear all,
I just commented on the TPC files. Can you confirm that I understand correctly:
- For **qcdb**, provenance (like "qc/", "qc_async/") is added automatically from the (to be provided) field `Activity.provenance`.
- For **ccdb**, provenance has to be provided in the path.
In this case, in the two places above, we either have to change to **qcdb** or put the provenance back.

File: Modules/TPC/run/tpcQCClusterVisualizer.json [hunk 1]
```diff
@@ -62,15 +62,15 @@
...
+        "path": "TPC/MO/Clusters/ClusterData",
         "dataType_comment": "This is the switch for 'RawDigits' or 'Clusters' task. Choose 'raw' or 'clusters'.",
         "dataType": "clusters",
         "initTrigger": [
           "once"
         ],
         "updateTrigger_comment": "To trigger on a specific file being updated, use e.g. 'newobject:ccdb:TPC/Calib/Noise'",
         "updateTrigger": [
-          "newobject:ccdb:qc/TPC/MO/Clusters/ClusterData"
+          "newobject:ccdb:TPC/MO/Clusters/ClusterData"
```
[the remark this answers is not in the document]
Comment (stheckel) [reply Modules/TPC/run/tpcQCClusterVisualizer.json:73]: Ah ok, so we were lucky. :)
Yes, please include it in the PR.

**Comment by knopers8** [comment]
> Dear Piotr, dear all, I just commented on the TPC files. Can you confirm that I understand correctly:
> 
>     * For **qcdb**, provenance (like "qc/", "qc_async/") is added automatically from the (to be provided) field `Activity.provenance`.
> 
>     * For **ccdb**, provenance has to be provided in the path.
> 
> 
> In this case, in the two places above, we either have to change to **qcdb** or put the provenance back.

Yes, you undestand correctly.

**Review by Barthelemy — APPROVED** [review]
That is very good, thank you

**Comment by Barthelemy** [comment]
@knopers8 I let you merge when you got hte feedback you needed.

**Comment by knopers8** [comment]
It seems I have green light from all of TPC, TOF and ITS, merging.

**Comment by knopers8** [comment]
On Monday 4 April (the expected FLP suite deploymend) we will have to adapt the config files in consul. I am putting a reminder to my calendar.

**Pull request merged**